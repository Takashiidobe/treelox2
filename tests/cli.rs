use std::process::Output;

use assert_cmd::cargo::CommandCargoExt;
use insta::{assert_yaml_snapshot, glob};
use insta_cmd::Command;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
struct TestOutput {
    status: i32,
    stdout: Vec<String>,
    stderr: Vec<String>,
}

#[test]
fn reference_files() {
    glob!("../test-files", "**/*.lox", |path| {
        let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap();

        let Output {
            status,
            stdout,
            stderr,
        } = cmd.arg(path).output().unwrap();

        let test_output = TestOutput {
            status: status.code().unwrap(),
            stdout: String::from_utf8_lossy(&stdout)
                .to_string()
                .lines()
                .map(|x| x.to_owned())
                .collect(),
            stderr: String::from_utf8_lossy(&stderr)
                .to_string()
                .lines()
                .map(|x| x.to_owned())
                .collect(),
        };

        assert_yaml_snapshot!(test_output);
    });
}
