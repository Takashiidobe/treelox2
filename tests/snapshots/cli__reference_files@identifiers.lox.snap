---
source: tests/cli.rs
expression: test_output
input_file: test-files/scanning/identifiers.lox
---
status: 0
stdout:
  - Identifier andy on line 1
  - Identifier formless on line 1
  - Identifier fo on line 1
  - Identifier _ on line 1
  - Identifier _123 on line 1
  - Identifier _abc on line 1
  - Identifier ab123 on line 1
  - Identifier abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890 on line 2
  - Identifier _ on line 2
  - Eof EOF on line 13
stderr: []
